stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -296,3 +313,37 @@ def is_greater_eq(value, other):
                     # compute how much packaging we can get
                     take = (need // pack_quantity) * pack_quantity
                     need = yield location, location_quantity, take, None, None
+
+    def _apply_strategy_by_lot(self, quants):
+        need = yield
+        product = fields.first(quants).product_id
+        comparison = {
+            "lot_qty_gt_reserved_qty": lambda lot_qty, need_qty: lot_qty > need_qty,
+            "lot_qty_gte_reserved_qty": lambda lot_qty, need_qty: lot_qty >= need_qty,
+            "lot_qty_eq_reserved_qty": lambda lot_qty, need_qty: lot_qty == need_qty,
+        }.get(self.reserve_lot)
+        if product.tracking == "lot" or comparison is not None:
+            for location, location_quants in quants._group_by_location():
```

Comment:
Can you clarify if your strategy is a single lot per location or a single lot ?
Location A; Lot2 5 units
Location B: Lot1 3 units + Lot2 4 units
If you need 8 units, do you want Lot1 from B & Lot2 from A. Or you want only Lot2 from B?

Reply:
@jbaudoux In this case no lots will be reserved, as it is not possible to reserve lots with qty < the requested qty.

Reply:
Ah yes, you're right. On purpose?

Reply:
then I don't understand why you group the quants by location

Reply:
@geomer198  please check it

Reply:
@geomer198 what about this?

Reply:
According to me, this needs to be addressed. In the other reservation rule, we try to select a location, that's why we group by location. As you try to select a single lot (no matter where), you should implement a `quants._group_by_lot()` keeping quants sorting (to respect removal strategy fifo/fefo). I dropped you a comment below on how to approach this.